### PR TITLE
[gui] Expandable runs

### PR DESCRIPTION
**Pull request by csordasmarton**

### CODE REVIEW

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 51
- Added 77
           

Complexity increasing per file
==============================
- web/server/vue-cli/src/mixins/date.mixin.js  1
- web/server/vue-cli/src/store/modules/run.js  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=6236530&bid=20319896)

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 51
- Added 78
           

Complexity increasing per file
==============================
- web/server/vue-cli/src/store/modules/run.js  9
- web/server/vue-cli/src/mixins/date.mixin.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=6236530&bid=20319896)

File: web/server/codechecker_server/api/report_server.py [hunk 1]
```diff
@@ -460,15 +460,26 @@ def process_run_history_filter(query, run_ids, run_history_filter):
...
-        query = query.filter(or_(*OR))
+        stored = run_history_filter.stored
+        if stored:
+            if stored.before:
+                stored_before = datetime.fromtimestamp(stored.before)
+                query = query.filter(RunHistory.time <= stored_before)
 
-    if run_history_filter and run_history_filter.tagIds:
-        query = query.filter(RunHistory.id.in_(run_history_filter.tagIds))
+            if stored.after:
```
[the remark this answers is not in the document]
Comment (csordasmarton) [reply web/server/codechecker_server/api/report_server.py:480]: @bruntib I tried it and the default value will be None if these fields are not set on the client side.

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 51
- Added 78
           

Complexity increasing per file
==============================
- web/server/vue-cli/src/mixins/date.mixin.js  1
- web/server/vue-cli/src/store/modules/run.js  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=6236530&bid=20319896)

**Review by gyorb — CHANGES_REQUESTED** [review]
Could you extend the history entries with some text to be more explicit about the numbers there.
So the users are not confused what those numbers mean (successful/failed analysis of the source files).
I think some users could think that those numbers are report numbers.

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 51
- Added 78
           

Complexity increasing per file
==============================
- web/server/vue-cli/src/mixins/date.mixin.js  1
- web/server/vue-cli/src/store/modules/run.js  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=6236530&bid=20319896)

**Comment by csordasmarton** [comment]
@gyorb I added a tooltip on these numbers and to this section.

I don't really want to add a string explicitly before the analyzer statistics section. By the way if I check a  run line I can see that these are analysis statistics, so I can expect that here these are also analyzer statistics. But with my changes if I hover my mouse over this it will shows me this information too.

What do you think?

**Review by gyorb — COMMENTED** [review]
Firefox breaks the tool tip lines a bit weird or is that appear only on my machine?
Like this:
```
Analysis statistics: shows the number of successfully analyzed
files
and the number of files which failed to analyze.
```
Chromium seems to be fine.

Otherwise LGTM.

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 51
- Added 128
           

Complexity increasing per file
==============================
- web/server/vue-cli/src/store/modules/run.js  9
- web/server/vue-cli/src/mixins/date.mixin.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=6236530&bid=20319896)

File: web/server/vue-cli/src/components/Run/AnalyzerStatisticsBtn.vue [hunk 2]
```diff
@@ -1,23 +1,39 @@
 <template>
-  <v-container
+  <span
     class="analyzer-statistics"
+    :title="'Analysis statistics: shows the number of successfully analyzed' +
```
Comment (gyorb) [inline web/server/vue-cli/src/components/Run/AnalyzerStatisticsBtn.vue:4]:
```suggestion
    :title="'Analysis statistics: shows the number of successfully analyzed ' +
```

**Review by gyorb — CHANGES_REQUESTED** [review]
a space is missing

**Review by gyorb — APPROVED** [review]

**Pull request merged**